org.jacoco.core/src/org/jacoco/core/internal/analysis/SourceNodeImpl.java

```diff
@@ -107,27 +107,30 @@ public void increment(final ISourceNode child) {
 	 *
 	 * @param instructions
 	 *            instructions to add
+	 * @param executions
+	 *            executions to add
 	 * @param branches
 	 *            branches to add
 	 * @param line
 	 *            optional line number or {@link ISourceNode#UNKNOWN_LINE}
 	 */
-	public void increment(final ICounter instructions, final ICounter branches,
-			final int line) {
+	public void increment(final ICounter instructions, final int executions,
```

Comment:
I think we call this method for every bytecode instruction. So the new executions counter is the total number of all executions of all instruction, and not the number of time the line was executed. Is that intended?

Reply:
Ah, yes. For my thesis I'm only interested in how often a line was executed. The quickest way to implement this was to propagate it along with these functions all the way down to `LineImpl#increment`. So in that regard, the Javadoc is a bit off here. I don't store the total number of executions on all instructions on a class/method level (but I could). 

For line executions, I don't sum all the instruction's executions but rather take the `max` again (see `LineImpl#increment` implementations). Summing would result in an indication that a line was executed more than it actually was. However, it also might give a wrong perception about how often specific source code was executed as some statements might be executed less than reported (which I will carefully document in my thesis :smile:)